mapper/mapper.go

```diff
@@ -1,203 +1,243 @@
 package mapper
 
 import (
-	"context"
-	"net/url"
-	"strconv"
-	"strings"
-
 	searchC "github.com/ONSdigital/dp-api-clients-go/site-search"
 	model "github.com/ONSdigital/dp-frontend-models/model/search"
 	"github.com/ONSdigital/dp-frontend-search-controller/data"
-	"github.com/ONSdigital/log.go/log"
 )
 
-// CreateSearchPage maps type cookies.Policy to model.Page
-func CreateSearchPage(ctx context.Context, query url.Values, respC searchC.Response, categories []data.Category) (page model.Page) {
+// CreateSearchPage maps type searchC.Response to model.Page
+func CreateSearchPage(validatedQueryParams data.SearchURLParams, categories []data.Category, respC searchC.Response) (page model.Page) {
 	// SEARCH STRUCT MAPPING
-	var err error
 	page.Metadata.Title = "Search"
 	page.SearchDisabled = true
-	page.Data.Query = query.Get("q")
 
-	page.Data.Filter.Query = query["filter"]
-	page.Data.Filter.Options = []string{"Publication", "Data", "Other"}
+	mapQuery(&page, validatedQueryParams, categories, respC)
 
-	page.Data.Sort.Query = query.Get("sort")
-	page.Data.Sort.LocaliseFilterKeys = getFilterSortKeyList(query, categories)
-	page.Data.Sort.FilterText = strings.Replace(strings.ToLower(page.Data.Sort.Query), "-", " ", 1)
+	mapResponse(&page, respC, categories)
 
-	if query.Get("limit") != "" {
-		page.Data.Limit, err = strconv.Atoi(query.Get("limit"))
-		if err != nil {
-			log.Event(ctx, "unable to convert search limit to int - default to limit 10", log.INFO)
-			page.Data.Limit = 10
-		}
-	}
+	return page
+}
 
-	if query.Get("offset") != "" {
-		page.Data.Offset, err = strconv.Atoi(query.Get("offset"))
-		if err != nil {
-			log.Event(ctx, "unable to convert search offset to int - default to offset 0", log.INFO)
-			page.Data.Offset = 0
-		}
+func mapQuery(page *model.Page, validatedQueryParams data.SearchURLParams, categories []data.Category, respC searchC.Response) {
+	page.Data.Query = validatedQueryParams.Query
+
+	page.Data.Filter = validatedQueryParams.Filter.Query
+
+	mapSort(page, validatedQueryParams, categories)
+
+	mapPagination(page, validatedQueryParams, categories, respC)
+}
+
+func mapSort(page *model.Page, validatedQueryParams data.SearchURLParams, categories []data.Category) {
```

Comment:
Categories is passed to this but it is never used, was it supposed to or shall we remove it?